tests/testthat/test-errorcapture.R

```diff
@@ -0,0 +1,168 @@
+##     params <- list(
+##       tweet.fields = "attachments,author_id,conversation_id,created_at,entities,geo,id,in_reply_to_user_id,lang,public_metrics,possibly_sensitive,referenced_tweets,source,text,withheld", 
+##       user.fields = "created_at,description,entities,id,location,name,pinned_tweet_id,profile_image_url,protected,public_metrics,url,username,verified,withheld", 
+##       expansions = "author_id,entities.mentions.username,geo.place_id,in_reply_to_user_id,referenced_tweets.id,referenced_tweets.id.author_id", 
+##       place.fields = "contained_within,country,country_code,full_name,geo,id,name,place_type"
+##     )
+##     endpoint_url <- "https://api.twitter.com/2/tweets"
+##     emptydir <- academictwitteR:::.gen_random_dir()  
+
+##     for (i in seq_len(3)) {
+##       index <- c(1, 21, 41)[i]
+##       index_end <- index+19
+##       x <- fff[index:index_end]
+##       params[["ids"]] <- paste0(x, collapse = ",")
+##       expect_error(new_rows <- get_tweets(params = params, endpoint_url = endpoint_url, n = Inf, file = NULL, bearer_token = get_bearer(), export_query = FALSE, data_path = emptydir, bind_tweets = TRUE, verbose = FALSE, errors = TRUE), NA)
+##       Sys.sleep(0.5)
+##       expect_equal(length(list.files(emptydir)), i * 3)
+##       expect_equal(length(list.files(emptydir, "^errors")), i)
+##       df <- bind_tweets(emptydir, verbose = FALSE)
+##       expect_equal(nrow(df), i * 20)
+##     }
+##     unlink(emptydir, recursive = TRUE)
+##   })
+
+## stop_capturing()
+
+with_mock_api({
+  test_that("normal case: errors = TRUE", {
+    skip_if(!dir.exists("api.twitter.com"))
+    fff <- readRDS("../testdata/fff_de.RDS")
+    params <- list(
+      tweet.fields = "attachments,author_id,conversation_id,created_at,entities,geo,id,in_reply_to_user_id,lang,public_metrics,possibly_sensitive,referenced_tweets,source,text,withheld", 
+      user.fields = "created_at,description,entities,id,location,name,pinned_tweet_id,profile_image_url,protected,public_metrics,url,username,verified,withheld", 
+      expansions = "author_id,entities.mentions.username,geo.place_id,in_reply_to_user_id,referenced_tweets.id,referenced_tweets.id.author_id", 
+      place.fields = "contained_within,country,country_code,full_name,geo,id,name,place_type"
+    )
+    endpoint_url <- "https://api.twitter.com/2/tweets"
+    emptydir <- academictwitteR:::.gen_random_dir()  
+
+    for (i in seq_len(3)) {
+      index <- c(1, 21, 41)[i]
+      index_end <- index+19
+      x <- fff[index:index_end]
+      params[["ids"]] <- paste0(x, collapse = ",")
+      expect_error(new_rows <- get_tweets(params = params, endpoint_url = endpoint_url, n = Inf, file = NULL, bearer_token = get_bearer(), export_query = FALSE, data_path = emptydir, bind_tweets = TRUE, verbose = FALSE, errors = TRUE), NA)
+      Sys.sleep(0.5)
+      expect_equal(length(list.files(emptydir)), i * 3)
+      expect_equal(length(list.files(emptydir, "^errors")), i)
```

Comment:
testing for `errors` json files even when `n_polluted` = 0 (no pollution, i.e. no error.)